### PR TITLE
Feature - sign in with material design and redirect

### PR DESCRIPTION
**Pull request by xgarrett**

### CODE REVIEW

File: src/actions/SignInActions.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { push as redirect } from 'react-router-redux';
 import * as types from 'constants/ActionTypes';
 import request from 'utils/requester';
```
Comment (xgarrett) [inline src/actions/SignInActions.js]: accidental, during rebase

File: src/components/SignInForm/index.js [hunk 2]
```diff
@@ -1,41 +1,94 @@
 import React, { Component, PropTypes } from 'react';
+import * as styles from './styles';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
```
Comment (FredZgred) [inline src/components/SignInForm/index.js]: You can import both in one line

File: src/components/SignInForm/index.js [hunk 3]
```diff
@@ -1,41 +1,94 @@
...
 
+        this.state = {
+            errors: {
+                login: '',
+                password: '',
+            },
+        };
+
         this.submit = this.submit.bind(this);
+        this.validate = this.validate.bind(this);
```
Comment (FredZgred) [inline src/components/SignInForm/index.js]: is this *bind* needed? You are not using validate method in html but inside class, so I think reference by *this* will be still possible without this bind method

File: src/containers/SignInView/index.js [hunk 4]
```diff
@@ -1,13 +1,15 @@
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
-import { signIn } from 'actions/SignInActions';
+import { signIn } from '../../actions/SignInActions';
```
Comment (FredZgred) [inline src/containers/SignInView/index.js]: hmm.. you were voting for absolute paths.. :)

File: src/containers/SignInView/index.js [hunk 5]
```diff
@@ -27,19 +29,24 @@ class SignInView extends Component {
...
-        const { hasUserData, triedSignIn } = this.props;
-        let signInMessage = null;
+        let signInError = null;
 
-        if (triedSignIn && hasUserData) {
-            signInMessage = 'Successful sign in!';
-        } else if (triedSignIn) {
-            signInMessage = 'Please correct your email/login or password.';
+        if (this.props.failedSignIn) {
+            signInError = 'Please correct your email/login or password.';
```
Comment (FredZgred) [inline src/containers/SignInView/index.js]: It is ok. You can also show here error message received from backend :) just small advice :)

File: src/index.scss [hunk 6]
```diff
@@ -3,3 +3,24 @@ $fa-font-path: "~font-awesome/fonts";
...
+
+body {
+    position: relative;
+    font-size: 1.6rem;
+}
+
+// style our app wrappers
+#root, [data-reactroot] {
+    height: 100%;
+}
```
Comment (FredZgred) [inline src/index.scss]: Chceck please, if all of those styles are needed here. I mean on the top normalize is included and maybe some of your styles are just duplication

File: src/reducers/UserReducer.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ import * as types from 'constants/ActionTypes';
 const initialState = {
     body: null,
     signingIn: false,
-    triedSignIn: false,
+    failedSignIn: false,
```
Comment (FredZgred) [inline src/reducers/UserReducer.js]: Wouldn't it be better to name it **signinFailed**? Just proposition

File: src/components/SignInForm/index.js (hunk 2 above)
Comment (xgarrett) [inline src/components/SignInForm/index.js]: :+1:

File: src/components/SignInForm/index.js (hunk 3 above)
Comment (xgarrett) [inline src/components/SignInForm/index.js]: you're right.

**Pull request merged**